packages/docs/site/docs/10-javascript-api/03-playground-api-client.md

```diff
@@ -70,6 +70,45 @@ await client.goTo('/test/index.php');
 
 For a full list of these methods, consult the PlaygroundClient interface.
 
+## Sending messages to JavaScript
+
+You can pass messages from PHP to JavaScript using the `post_message_to_js()` function. It accepts two arguments:
+
+-   `$type` – is used to call the correct listener
+-   `$data` – any extra information as a string
+
+For example, here's how you would send a message called "post-save" with a JSON-encoded post ID and title:
+
+```ts
+const php = await PHP.load('8.0');
+
+php.onMessage(
+    'post-save',
```

Comment:
what's the point of the `$type` here? normal `postMessage` messages are just data and all feed to the same place, so why create a name?

this is closer to a `MessagePort` channel, though still different.

Reply:
This is not strictly needed. I thought everyone will end up doing that on their own though and put it in the library instead. However, this API could only accept a string and then have everyone do their own routing in the event handler. What do you think?

Reply:
given that _no_ JS post message interface has a name argument I lean towards omitting it. we're either dispatching here and designating that these channels need a string name or we let app code do it however they want.

if a single app needs to dispatch based on more than the name (on some value, for instance), then we've already defeated the help we offer by providing this.

actually I think this is also part of what corresponds well to `.onmessage = ` because there's some suggestion that there's a single handler (even though `.addEventListener('message', …)` is also fine.

Reply:
Thanks @dmsnell, I think you're right about the parameter. I just removed it. `onmessage = ` or `.addEventListener()` will make a good follow-up, but for now let's land this to unblock @ellatrix since – it's not adding a new convention, only following the one that's already there. The deprecation for on* will still need to happen.